src/GovUk.Education.ExploreEducationStatistics.Notifier/Functions/ReleaseNotifier.cs

```diff
@@ -0,0 +1,204 @@
+        _emailService = emailService ?? throw new ArgumentNullException(nameof(emailService));
+        _storageTableService = storageTableService ?? throw new ArgumentNullException(nameof(storageTableService));
+        _configurationProvider = configurationProvider
+                                 ?? throw new ArgumentNullException(nameof(configurationProvider));
+        _notificationClientProvider = notificationClientProvider
+                                 ?? throw new ArgumentNullException(nameof(notificationClientProvider));
+    }
+
+    [FunctionName("ReleaseNotifier")]
+    // ReSharper disable once UnusedMember.Global
+    public async Task ReleaseNotifierFunc(
+        [QueueTrigger(ReleaseNotificationQueue)]
+        ReleaseNotificationMessage msg,
+        ILogger logger,
+        ExecutionContext context)
+    {
+        logger.LogInformation("{FunctionName} triggered",
+            context.FunctionName);
+
+        var config = _configurationProvider.Get(context);
+
+        var notifyApiKey = config.GetValue<string>(NotifyApiKeyName);
+        var notificationClient = _notificationClientProvider.Get(notifyApiKey);
+
+        var storageConnectionStr = config.GetValue<string>(StorageConnectionName);
+        var subscribersTable = await _storageTableService.GetTable(storageConnectionStr, SubscriptionsTblName);
+
+        var sentEmails = new HashSet<string>();
+
+        // Send emails to subscribers of publication
+        var releaseSubscriberQuery = new TableQuery<SubscriptionEntity>()
+            .Where(TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.Equal,
+                msg.PublicationId.ToString()));
+        var releaseSubscriberEmails = await GetSubscriberEmails(subscribersTable, releaseSubscriberQuery);
+
+        foreach (var email in releaseSubscriberEmails)
+        {
+            SendSubscriberEmail(email, msg, notificationClient, config);
+            sentEmails.Add(email);
+        }
+
+        logger.LogInformation("Emailed {NumReleaseSubscriberEmailsSent} publication subscribers",
+            sentEmails.Count);
+
+        // Send emails to subscribers of any associated superseded publication
+        foreach (var supersededPublication in msg.SupersededPublications)
+        {
+            var releaseSupersededPubSubsQuery = new TableQuery<SubscriptionEntity>()
+                .Where(TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.Equal,
+                    supersededPublication.Id.ToString()));
+            var supersededPublicationSubscriberEmails = await GetSubscriberEmails(
+                subscribersTable, releaseSupersededPubSubsQuery);
+
+            var numSupersededSubscriberEmailsSent = 0;
+
+            foreach (var email in supersededPublicationSubscriberEmails)
+            {
+                if (sentEmails.Contains(email))
+                {
+                    continue;
+                }
```

Comment:
Do you know if all subscriber email addresses are held in a certain case, because this check will be case sensitive resulting in subscribers potentially getting more than one email if so.

Reply:
Good catch.

Discovered you can just do this: `new HashSet<string>(StringComparer.OrdinalIgnoreCase);`